code/modules/mob/living/basic/icemoon/ice_whelp/ice_whelp_ai.dm

```diff
@@ -0,0 +1,217 @@
+	. = ..()
+	controller.clear_blackboard_key(target_key)
+
+///subtree to find icy rocks and create sculptures out of them
+/datum/ai_planning_subtree/sculpt_statues
+
+/datum/ai_planning_subtree/sculpt_statues/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	var/obj/target = controller.blackboard[BB_TARGET_ROCK]
+
+	if(QDELETED(target))
+		controller.queue_behavior(/datum/ai_behavior/find_and_set, BB_TARGET_ROCK, /obj/structure/flora/rock/icy)
+		return
+
+	controller.queue_behavior(/datum/ai_behavior/sculpt_statue, BB_TARGET_ROCK)
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+/datum/ai_behavior/sculpt_statue
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_REQUIRE_REACH | AI_BEHAVIOR_CAN_PLAN_DURING_EXECUTION
+	action_cooldown = 5 MINUTES
+
+/datum/ai_behavior/sculpt_statue/setup(datum/ai_controller/controller, target_key)
+	. = ..()
+	var/obj/target = controller.blackboard[target_key]
+	if(QDELETED(target))
+		return FALSE
+	set_movement_target(controller, target)
+
+/datum/ai_behavior/sculpt_statue/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
+	. = ..()
+
+	var/atom/target = controller.blackboard[target_key]
+	var/mob/living/basic/mining/ice_whelp/living_pawn = controller.pawn
+
+	if(QDELETED(target))
+		finish_action(controller, FALSE, target_key)
+		return
+
+	living_pawn.create_sculpture(target)
+	finish_action(controller, TRUE, target_key)
+
+/datum/ai_behavior/sculpt_statue/finish_action(datum/ai_controller/controller, succeeded, target_key)
+	. = ..()
+	controller.clear_blackboard_key(target_key)
+
+
+//subtree to use our attacks on the victim
+/datum/ai_planning_subtree/targeted_mob_ability/ice_whelp
+	ability_key = BB_WHELP_STRAIGHTLINE_FIRE
+	use_ability_behaviour = /datum/ai_behavior/targeted_mob_ability/ice_whelp
+	finish_planning = FALSE
+
+
+/datum/ai_behavior/targeted_mob_ability/ice_whelp
+	///key that stores how enraged we are
+	var/enraged_key = BB_WHELP_ENRAGED
+	///key that stores the ability we will use instead if we are fully enraged
+	var/secondary_ability_key = BB_WHELP_WIDESPREAD_FIRE
+
+/datum/ai_behavior/targeted_mob_ability/ice_whelp/perform(seconds_per_tick, datum/ai_controller/controller, ability_key, target_key)
+	var/datum/action/cooldown/ability
+	var/enraged_value = controller.blackboard[enraged_key]
+
+	if(prob(enraged_value))
+		ability = controller.blackboard[secondary_ability_key]
+		controller.set_blackboard_key(enraged_key, 0)
+	else
+		ability = controller.blackboard[ability_key]
+		controller.set_blackboard_key(enraged_key, enraged_value + 25)
+
+	var/mob/living/target = controller.blackboard[target_key]
+	if(QDELETED(ability) || QDELETED(target))
+		finish_action(controller, FALSE, ability_key, target_key)
+		return
+
+	var/mob/pawn = controller.pawn
+	var/result = ability.InterceptClickOn(pawn, null, target)
+	finish_action(controller, result, ability_key, target_key)
```

Comment:
This is copy pasted from the parent call from what I can see - which means there's likely a better way to go about implementing this subtype without copying perform

IE, replace `ability = controller.blackboard[ability_key]` with something like `abliity = get_ability_to_use(controller, ability_key)` and implement that proc so you can override it